c7n/executor.py

```diff
@@ -40,17 +40,29 @@ def executor(name, **kw):
 
 
 class MainThreadExecutor(object):
+    """ For running tests.
+
+    async == True  -> catch exceptions and store them in the future.
+    async == False -> let exceptions bubble up.
+    """
+    
     # For Dev/Unit Testing with concurrent.futures
-    def __init__(self, *args, **kw):
+    def __init__(self, async=True, *args, **kw):
```

Comment:
goal/ask was this would be a class attribute as else it alters the interface for concurrent.futures and we don't currently allow for customization of the callers without modification.